modules/afsocket/afsocket-source.c

```diff
@@ -749,9 +752,7 @@ static void
 afsocket_sd_start_watches(AFSocketSourceDriver *self)
 {
   _listen_fd_start(self);
-
-  if (self->dynamic_window_pool != NULL)
-    _dynamic_window_timer_start(self);
+  _dynamic_window_timer_start(self);
```

Comment:
I think it is cleaner if `_dynamic_window_timer_start()` actually starts the timer as its name suggests. It is now more of a `_dynamic_window_timer_maybe_start()`, which sounds mystic.

Reply:
I've used the "start" methods similar to how init() is assumed to work, e.g. initialize that portion of the driver, in a similar vein "stop" is parallel do deinit().

The issue is that init is taken and that initializes the low-level parts of the ivykis structs. I didn't change this yet, so keeping this comment open.

Reply:
Sorry, I don't understand this.
`init()` and `deinit()` does initialization and deinitialization unconditionally.

My comment was a clean-code-related note about a refactor step that made `_dynamic_window_timer_start()` conditional. The function is called timer_start, but after this commit, it sometimes does not start the timer.